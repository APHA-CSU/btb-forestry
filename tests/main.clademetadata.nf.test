nextflow_process {

    name "Test Process cladeMetadata"
    script "main.nf"
    process "cladeMetadata"

    test("Should process clade metadata") {

        when {
            params {
                today = "01Jan25"
            }
            process {
                """
                input[0] = [ 'test', 
                file("${projectDir}/tests/data/samplelist.csv"), 
                file("${projectDir}/tests/data/filterMeta_exp.csv")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
