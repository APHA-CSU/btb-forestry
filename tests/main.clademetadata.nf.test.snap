{
    "Should process clade metadata": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "test_metadata_01Jan25.csv:md5,99479c045f65a51346a78c98d53de61d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T14:05:54.589108515"
    }
}